apps/prairielearn/src/batched-migrations/20240523183853_sync_courses_for_implicit_flag.ts

```diff
@@ -0,0 +1,64 @@
+import { makeBatchedMigration } from '@prairielearn/migrations';
+import * as namedLocks from '@prairielearn/named-locks';
+import { queryOneRowAsync, queryRows } from '@prairielearn/postgres';
+
+import { Course, CourseSchema } from '../lib/db-types.js';
+import { createServerJob } from '../lib/server-jobs.js';
+import { getLockNameForCoursePath } from '../models/course.js';
+import { syncDiskToSqlWithLock } from '../sync/syncFromDisk.js';
+
+export default makeBatchedMigration({
+  async getParameters() {
+    const result = await queryOneRowAsync('SELECT MAX(id) as max from pl_courses;', {});
+    return {
+      min: 1n,
+      max: result.rows[0].max,
+      batchSize: 10,
+    };
+  },
+
+  async execute(min: bigint, max: bigint): Promise<void> {
+    const courses = await queryRows(
+      'SELECT * FROM pl_courses WHERE id >= $min AND id <= $max AND deleted_at IS NULL',
+      { min, max },
+      CourseSchema,
+    );
+
+    for (const course of courses) {
+      await syncCourse(course);
+    }
+  },
+});
+
+/**
+ * Re-syncs an existing course. Does NOT pull new changes from the remote repository.
+ */
+export async function syncCourse(course: Course) {
```

Comment:
If we find ourselves needing to write more background migrations that re-sync courses, we can move this to e.g. `lib/courses`, but I figured it's fine to keep it here for now.